api/v1beta1/cluster_types.go

```diff
@@ -122,6 +122,11 @@ type ControlPlaneTopology struct {
 	// +optional
 	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
 
+	// NodeVolumeDetachTimeout is the total amount of time that the controller will spend on waiting for all volumes
+	// to be detached. The default value is 0, meaning that the volumes can be detached without any time limitations.
+	// +optional
+	NodeVolumeDetachTimeout *metav1.Duration `json:"nodeVolumeDetachTimeout,omitempty"`
+
```

Comment:
Please also do the same for MachineDeploymentTopology + corresponding changes in other files